### PR TITLE
Add an example sidecar calculator for functional tests

### PR DESCRIPTION
**Pull request by Barakmor1**
**What this PR does / why we need it**:
Add an example sidecar calculator for functional tests and API exercise.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Barakmor1** [comment]
I added all vendors of the example sidecar to a separate commit, which is the second commit of this PR, so it will be easier to review.

File: example_sidecars/label-sidecar/cmd/label-sidecar.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	labelAppLabel            = "label-app"
+)
+
+type configName string
+
+type labelCalculator struct {
+	evaluatorConfig configName
+}
+
+func (lc *labelCalculator) PodUsageFunc(podToEvaluate *corev1.Pod, _ []*corev1.Pod) (corev1.ResourceList, error, bool) {
```
Comment (iholder101) [inline example_sidecars/label-sidecar/cmd/label-sidecar.go:26]: It's feels a bit odd to have the error return param at the end. We can change the signature to the following in a follow-up PR:
`PodUsageFunc(podToEvaluate *corev1.Pod, _ []*corev1.Pod) (corev1.ResourceList, bool, error)`

Another alternative is to provide a known error value to specify that the pod is not affected, therefore removing the bool argument:
```go
func (lc *labelCalculator) PodUsageFunc(podToEvaluate *corev1.Pod, _ []*corev1.Pod) (corev1.ResourceList, error) {
	if something {
		return nil, libsidecar.errPodNotAffected
	}
}
```

File: example_sidecars/label-sidecar/cmd/label-sidecar.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// Triple Calculate triple amount of usage for pod
+	Triple        configName = "triple"
+	labelAppLabel            = "label-app"
+)
+
+type configName string
+
+type labelCalculator struct {
+	evaluatorConfig configName
+}
```
Comment (iholder101) [inline example_sidecars/label-sidecar/cmd/label-sidecar.go:24]: You can add the following to make sure the struct is compliant with the interface:
```go
var _ = libsidecar.SidecarCalculator(&labelCalculator{})
```

File: tests/e2e_plugable_policies_test.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+		BeforeEach(func() {
+			currAAQ, err := utils.GetAAQ(f)
+			Expect(err).ToNot(HaveOccurred())
+			if libaaq.CheckIfPlugablePolicyExistInAAQ(currAAQ, libaaq.LabelSidecar, libaaq.Double) {
+				return
+			}
+			Eventually(func() error {
+				aaq, err := utils.GetAAQ(f)
+				Expect(err).ToNot(HaveOccurred())
+				aaq.Spec.Configuration.SidecarEvaluators = []v1.Container{}
```
Comment (iholder101) [inline tests/e2e_plugable_policies_test.go:34]: seems redundant

**Review by iholder101 — COMMENTED** [review]
Looks really cool!!

File: example_sidecars/label-sidecar/cmd/label-sidecar.go (hunk 1 above)
Comment (Barakmor1) [reply example_sidecars/label-sidecar/cmd/label-sidecar.go:26]: Yes, you're absolutely right about that one. I will make sure to add another PR before this one gets merged.

Comment (Barakmor1) [reply example_sidecars/label-sidecar/cmd/label-sidecar.go:26]: Here:
https://github.com/kubevirt/application-aware-quota/pull/56#issuecomment-2081421947

Comment (Barakmor1) [reply example_sidecars/label-sidecar/cmd/label-sidecar.go]: Done :)

File: example_sidecars/label-sidecar/cmd/label-sidecar.go (hunk 2 above)
Comment (Barakmor1) [reply example_sidecars/label-sidecar/cmd/label-sidecar.go:24]: Done :)

File: tests/e2e_plugable_policies_test.go (hunk 3 above)
Comment (Barakmor1) [reply tests/e2e_plugable_policies_test.go]: Right, removed.

**Comment by iholder101** [comment]
/lgtm
Thank you!

**Pull request merged**